src/pages/my_registrations/MyRegistrations.tsx

```diff
@@ -44,6 +54,50 @@ export const MyRegistrations = ({ selectedUnpublished, selectedPublished }: MyRe
       return 0;
     });
 
+  const draftRegistrationMutation = useMutation({
+    mutationFn: () => deleteAllDraftRegistrations(),
+    onSuccess: () => {
+      dispatch(
+        setNotification({
+          message: t('feedback.success.delete_registration'),
+          variant: 'success',
+        })
+      );
+      registrationsQuery.refetch();
+    },
+    onError: () =>
+      dispatch(
+        setNotification({
+          message: t('feedback.error.delete_registration'),
+          variant: 'error',
+        })
+      ),
+  });
+
+  const deleteAllDraftRegistrations = async () => {
+    setShowDeleteModal(true);
+
+    setIsDeleting(true);
+
+    const deletePromises = draftRegistrations.map(async (registration) => {
+      const identifierToDelete = getIdentifierFromId(registration.id);
+      await deleteRegistration(identifierToDelete);
+    });
+
+    try {
+      await Promise.all(deletePromises);
+    } catch (error) {
+      dispatch(setNotification({ message: t('feedback.error.delete_registration'), variant: 'error' }));
+    } finally {
+      setIsDeleting(false);
+      setShowDeleteModal(false);
+    }
```

Comment:
useMutation kan sikkert håndtere dette?
```suggestion
    await Promise.all(deletePromises);
```